src/main/java/org/ow2/proactive/catalog/service/CatalogObjectCallGraphService.java

```diff
@@ -52,15 +57,33 @@ public class CatalogObjectCallGraphService {
     @Autowired
     private CatalogObjectCallGraphPDFGenerator catalogObjectCallGraphPDFGenerator;
 
-    public byte[] generateBytesCallGraph(List<String> authorisedBucketsNames, Optional<String> kind,
+    public byte[] generateBytesCallGraphForAllBucketsAsZip(List<String> authorisedBucketsNames, Optional<String> kind,
             Optional<String> contentType) {
 
-        List<CatalogObjectMetadata> metadataList = catalogObjectService.listCatalogObjects(authorisedBucketsNames,
-                                                                                           kind,
-                                                                                           contentType);
-
-        return catalogObjectCallGraphPDFGenerator.generatePdfImage(metadataList, kind, contentType);
-
+        Map<String, List<CatalogObjectMetadata>> bucketNameCatalogObjectMetadata = catalogObjectService.listCatalogObjects(authorisedBucketsNames,
```

Comment:
byteArrayOutputStream.toByteArray() call will create a copy of the entire output in memory, which can be problematic if we have very large outputs. Would consider streaming the data or processing the PDFs incrementally.

Reply:
I agree it would be better but my goal wasn't to refactor the PDF generation process but just update the result to fix the feature. I think that creating the PDF incrementally would require quite the refactoring.
The threshold here is more than 700 objects in a single bucket.